packages/pages/src/components/analytics/hooks.ts

```diff
@@ -0,0 +1,102 @@
+import { ConversionDetails, Visitor } from "@yext/analytics";
+import { useContext } from "react";
+import { AnalyticsContext } from "./context";
+import { AnalyticsMethods } from "./interfaces";
+
+/**
+ * The useAnalytics hook can be used anywhere in the tree below a configured
+ * AnalyticsProvider.  Calling it will return an object to give you access to
+ * the analytics convenience methods for use in your components,
+ * such as track(), pageView(), optIn() etc.
+ *
+ * @public
+ */
+export function useAnalytics(): AnalyticsMethods {
+  const ctx = useContext(AnalyticsContext);
+  if (!ctx) {
+    throw new Error(
+      "Attempted to call useAnalytics outside of an AnalyticsProvider"
+    );
+  }
+
+  // TODO: is this the right way / place to expose a callback for use by a Cookie Management banner?
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  if (!window.setAnalyticsOptIn) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    window.setAnalyticsOptIn = () => {
+      ctx.optIn();
+    };
+  }
+
+  // TODO: this is ugly, I imagine there is a more elegant way of doing this
+  return {
+    trackClick(
+      eventName: string,
+      conversionData?: ConversionDetails
+    ): (e: MouseEvent) => Promise<void> {
+      return ctx.trackClick(eventName, conversionData);
+    },
+    setDebugEnabled(enabled: boolean): void {
+      return ctx.setDebugEnabled(enabled);
+    },
+    enableTrackingCookie(): void {
+      return ctx.enableTrackingCookie();
+    },
+    identify(visitor: Visitor): void {
+      return ctx.identify(visitor);
+    },
+    optIn(): Promise<void> {
+      return ctx.optIn();
+    },
+    pageView(): Promise<void> {
+      return ctx.pageView();
+    },
+    pushScope(name: string): void {
+      return ctx.pushScope(name);
+    },
+    track(
+      eventName: string,
+      conversionData?: ConversionDetails
+    ): Promise<void> {
+      return ctx.track(eventName, conversionData);
+    },
+  };
+}
+
+/**
+ * Simpler hook that just returns the analytics track() method.
+ *
+ * @public
+ */
+export const useTrack = () => {
+  return useAnalytics().track;
+};
+
+/**
+ * Simpler hook that just returns returns the analytics pageView method
+ *
+ * @public
+ */
+export const usePageView = () => {
+  return useAnalytics().pageView;
+};
+
+/**
+ * Simpler hook that just returns the analytics identify method
+ *
+ * @public
+ */
+export const useIdentify = () => {
+  return useAnalytics().identify;
+};
+
+/**
+ * Simpler hook that just returns the analytics pushScope method
+ *
+ * @public
+ */
+export const useScope = () => {
+  return useAnalytics().pushScope;
```

Comment:
maybe it makes more sense to have a `setScope` than `pushScope`? I think the imperative way pushscope works gets confusing in the react components. does it push every time the component rerenders?

Reply:
that is a good question that i need to test